test/Maps/MapReader.test.cpp

```diff
@@ -0,0 +1,24 @@
+#include "Maps/MapReader.h"
+#include "Maps/MapType.h"
+#include "gtest/gtest.h"
+#include <string>
+
+TEST(MapReader, EmptyFilename) {
+	EXPECT_THROW(MapReader::Read(std::string(), MapType::Map), std::runtime_error);
+	EXPECT_THROW(MapReader::Read(std::string(), MapType::Save), std::runtime_error);
+}
+
+TEST(MapReader, MissingFile) {
+	EXPECT_THROW(MapReader::Read("MissingFile.map", MapType::Map), std::runtime_error);
+	EXPECT_THROW(MapReader::Read("MissingFile.op2", MapType::Save), std::runtime_error);
+}
+
+//TEST(MapReader, EmptyFile) {
+//	EXPECT_THROW(MapReader::Read("Maps/data/EmptyMap.map", MapType::Map), std::runtime_error);
+//	EXPECT_THROW(MapReader::Read("Maps/data/EmptySave.OP2", MapType::Save), std::runtime_error);
+//}
+
+//TEST(MapReader, WrongExtension) {
+//	EXPECT_THROW(MapReader::Read("Maps/data/WrongExtension.txt", MapType::Map), std::runtime_error);
+//	EXPECT_THROW(MapReader::Read("Maps/data/WrongExtension.txt", MapType::Save), std::runtime_error);
+//}
```

Comment:
No, I don't think we should be checking for a bad extension, nor throwing in that case.

Reply:
Sounds good, will remove